src/Workspaces/CSharp/Portable/LanguageServices/CSharpTypeInferenceService.TypeInferrer.cs

```diff
@@ -1505,12 +1506,54 @@ private IEnumerable<TypeInferenceInfo> GetPatternTypes(PatternSyntax pattern)
             {
                 switch (pattern)
                 {
-                    case DeclarationPatternSyntax declarationPattern: return GetTypes(declarationPattern.Type);
                     case ConstantPatternSyntax constantPattern: return GetTypes(constantPattern.Expression);
+                    case DeclarationPatternSyntax declarationPattern: return GetTypes(declarationPattern.Type);
+                    case DeconstructionPatternSyntax deconstructionPattern: return GetTypesForDeconstructionPattern(deconstructionPattern);
+                    case PropertyPatternSyntax propertyPattern: return GetTypes(propertyPattern.Type);
                     default: return SpecializedCollections.EmptyEnumerable<TypeInferenceInfo>();
                 }
             }
 
+            private IEnumerable<TypeInferenceInfo> GetTypesForDeconstructionPattern(DeconstructionPatternSyntax deconstructionPattern)
+            {
+                // if it's of the for "X (...)" then just infer 'X' as the type.
```

Comment:
A `DeconstructionPatternSyntax` is a `PatternSyntax`. I think the first case would be covered by `GetTypeInfo(deconstructionPattern).ConvertedType`.

Reply:
Done.